transcript/src/main/java/com/transcriptanalyzer/transcript/User_Requests_Intents/Service/Tree.java

```diff
@@ -0,0 +1,108 @@
+package com.transcriptanalyzer.transcript.User_Requests_Intents.Service;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+// Assume we are producing one tree for a certain transcript in the ArrayList of transcripts
+public class Tree {
+    private Node treeRootNode;
+    private HashMap<String, Integer> counts;
+
+    // class for tree representing intents as it's nodes
+    // intents organized in order of their turn in their respective transcript
+    public Tree(ArrayList<ArrayList<ArrayList<String>>> transcripts) {
+        // root of tree
+        this.treeRootNode = new Node("Root", null);
+        Node start = this.treeRootNode;
+        this.counts = new HashMap<String, Integer>();
+
+        // go through each transcript, then each turn (representing each message intent pair)
+        // and add it as a subIntent node of the treeNode, then update treeNode down
+        for (ArrayList<ArrayList<String>> transcript : transcripts) {
+            for (ArrayList<String> turn : transcript) {
+                if (turn.size() == 2) {
+                    String currIntent = turn.get(1);
+                    // dont create node if it's none
+                    if (!Objects.equals(currIntent, "None")) {
+                        Node newRoot = this.treeRootNode.addSubIntent(currIntent);
+                        // add intent if doesnt exist, increment existing count if exists
+                        this.counts.merge(newRoot.intent, 1, Integer::sum);
+                        this.treeRootNode = newRoot;
+                    }
+                }
```

Comment:
Consider extracting this into a separate helper method to remove the nesting and improve code cleanliness - you should be able to just highlight these lines, right-click, and there should be an option to extract the method!